lib/src/component_declaration/component_base_2.dart

```diff
@@ -187,32 +188,97 @@ abstract class UiComponent2<TProps extends UiProps> extends react.Component2
-  /// For performance reasons, propTypes is only checked in development mode.
+  /// Allows usage of [react.PropValidator] functions to check the validity of a prop within the props passed to it.
   ///
   /// Override with a custom implementation to easily add validation.
   ///
-  ///     get propTypes => {
-  ///           getPropKey((props) => props.twoObjects, typedPropsFactory):
-  ///               (props, propName, componentName, location, propFullName) {
-  ///             final length = props.twoObjects?.length;
-  ///             if (length != 2) {
-  ///               return PropError.value(length, propName, 'must have a length of 2');
-  ///             }
-  ///             return null;
-  ///           },
-  ///         };
+  /// > For performance reasons, `propTypes` is only checked in development mode.
   ///
-  /// `getPropKey` is a statically typed helper to get the string key for a prop.
+  /// __Example of enforcing the length of a prop value:__
   ///
-  /// __Note:__ An improved version of `getPropKey` will be offered once
-  /// https://jira.atl.workiva.net/browse/CPLAT-6655 is completed.
+  /// ```
+  /// @Props()
+  /// class MyProps extends UiProps {
+  ///   Object foo;
+  /// }
   ///
-  /// For more info see: https://www.npmjs.com/package/prop-types
+  /// @Component2()
+  /// class MyComponent extends UiComponent2<MyProps> {
+  ///   @override
+  ///   get propTypes => {
+  ///     getPropKey((p) => p.foo): (props, info) {
+  ///       final length = props.foo?.length;
+  ///       if (length != 2) {
+  ///         return PropError.value(length, info.propName, 'must have a length of 2');
+  ///       }
+  ///       return null;
+  ///     },
+  ///   };
+  /// }
+  /// ```
+  ///
+  /// * [getPropKey] - as shown in the example above - is a statically typed helper to get the string key for a prop.
+  /// * The second argument (`info`) of the function used to return a value for the key contains metadata about
+  ///   the prop specified by the key.
+  ///     * `propName`, `componentName`, `location` and `propFullName` are available.
+  ///
+  /// __Example of enforcing the relationship between two prop values:__
+  ///
+  /// ```
+  /// @Props()
+  /// class MyProps extends UiProps {
+  ///   bool mustHaveAnotherPropValue;
+  ///   String anotherProp;
+  /// }
+  ///
+  /// @Component2()
+  /// class MyComponent extends UiComponent2<MyProps> {
+  ///   @override
+  ///   get defaultProps => newProps()..someProp = false;
+  ///
+  ///   @override
+  ///   get propTypes => {
+  ///     getPropKey((p) => p.mustHaveAnotherPropValue): (props, info) {
+  ///       if (props.mustHaveAnotherPropValue && props.anotherProp == null) {
+  ///         return PropError.combination(info.propName, 'anotherProp',
+  ///             'must have a non-null value when ${info.propName} is true.');
+  ///       }
+  ///       return null;
```

Comment:
`return null` is no longer part of the boilerplate
```suggestion
```

Reply:
https://github.com/cleandart/react-dart/pull/228#discussion_r343752079